docs/install-alternative.md

```diff
@@ -17,19 +17,81 @@ Follow the instructions below, or [watch a video by Drupalize.me](https://youtu.
-
 * [Documentation for composer's require command.](http://getcomposer.org/doc/03-cli.md#require)
 * Uninstall with : `composer global remove drush/drush`
 
+Install Drush for all users via Composer
+------------
+If you need a common Drush install available for all users on a system, follow the composer install steps below.
+
+1. [Install Composer globally][composer install global].
+1. Create and/or navigate to a directory path for the single composer drush install. In this example we are going to install drush version 7.x
+
+        sudo mkdir --parents /opt/drush/7.x
+        cd /opt/drush/7.x
+
+1. Initialise a new composer project that requires drush. Here you may also specify the drush [package version][composer package version] you wish composer to install. In this example we specify drush version 7.x with the [package version][composer package version] string `7.*`
+
+        sudo composer init --require=drush/drush:7.* -n
+
+1. Configure the path composer should use for the drush package's [vendor binaries][composer vendor binaries] or command-line scripts. Choose a directory path that is used in the `$PATH` configuration for all users, for example `/usr/local/bin`:
+
+        sudo composer config bin-dir /usr/local/bin
+
+1. Now run the `composer install` command. A `composer.json` file containing each of the configurations we just did has been created.
+
+        sudo composer install
+
+    The `composer.json` file tells composer to install (or update) drush at the [package version][composer package version] we specified and to put the [vendor binaries][composer vendor binaries] where all users can access them.
+
+1. Lastly, if you are using bash with command completion you can enable drush completion for all users by simply symlinking the `drush.complete.sh` shell script into the correct location.
+
+        sudo ln -s /usr/local/bin/drush.complete.sh \
+          /etc/bash_completion.d/drush
+
+**Important Tip:** When installing *drush 6.x* there are *dependencies not managed by composer* that will require download on first run. You
+should execute drush as a privelged user once after install (run `sudo drush --version`) to allow download of those libraries.
+
+### Getting Updates
+
+Use composer to update the drush library just as you would with any other composer managed project.
+
+1. Navigate to the drush install directory path. This is the path where your `composer.json` file for the drush install was created.
+
+        cd /opt/drush/7.x
+
+1. Run composer update
+
+        sudo composer update
+
+After composer update completes, the binaries and library will be updated for all users.
+
+### Major Version Upgrade
+
+If upgrading to a new major version the steps are the same, with one addition:
+
+1. Remove the existing symlinks to drush [vendor binaries][composer vendor binaries]. From our example above these would be located in the `bin-dir` path `/usr/local/bin`:
+
+        sudo rm -i /usr/local/bin/drush*
```

Comment:
I wish Composer had a command to uninstall the items it added to the bin directory. Beyond this one shortcoming, I am overall positively inclined towards these instructions.

Some folks advise against `sudo drush`, and `sudo composer` has similar characteristics vis-a-vis custom installers. However, the basic procedure here involves installing code to a directory that is the the global PATH for all users, so I suppose that adding `sudo` does not add a lot of extra exposure over that. It would complicate the instructions a lot recommend creating a `composer` user, making a separate Composer bin directory, adding that to the PATH, etc. etc., so it's probably best to stay with the current form.

Is `sudo composer` and the global bin technique described being used elsewhere now? The comments below imply that perhaps this is becoming a general use pattern.